osf/models/notable_domain.py

```diff
@@ -40,6 +45,56 @@ def __repr__(self):
     def __str__(self):
         return repr(self)
 
+    @staticmethod
+    def check_resource_for_domains(obj, confirm_spam=False, send_to_moderation=False):
+        fields_to_query = []
+        for field in list(obj.SPAM_CHECK_FIELDS):
+            fields_to_query += [models.Q(**{f'{field}__regex': settings.DOMAIN_REGEX})]
+        query = reduce(operator.or_, fields_to_query)
+        has_domain = obj.__class__.objects.filter(query).filter(id=obj.id).exists()
+
+        if not has_domain:
+            return False
+        elif confirm_spam and NotableDomain.has_spam_domain(obj):
+            obj.confirm_spam(save=True)
+        elif send_to_moderation:
+            NotableDomain.add_domains_to_moderation_queue(obj)
+        obj.save()
+        return True
+
+    @staticmethod
+    def add_domains_to_moderation_queue(obj):
+        fields_to_query = []
+        spam_fields = list(obj.SPAM_CHECK_FIELDS)
+        for field in spam_fields:
+            fields_to_query += [models.Q(**{f'{field}__regex': settings.DOMAIN_REGEX})]
+        query = reduce(operator.or_, fields_to_query)
+        raw_field_content = str(obj.__class__.objects.filter(query).filter(id=obj.id).values(*spam_fields))
+
+        urls = re.findall(settings.DOMAIN_REGEX, raw_field_content)
+        for url in urls:
+            domain = urlparse(url)
+            notable_domain, created = NotableDomain.objects.get_or_create(
+                domain=f'{domain.scheme}://{domain.netloc}',  # remove path and query params
+                defaults={'note': NotableDomain.Note.UNKNOWN}
+            )
+            DomainReference.objects.get_or_create(
+                domain=notable_domain,
+                referrer_object_id=obj.id,
+                referrer_content_type=ContentType.objects.get_for_model(obj)
+            )
+
+    @staticmethod
+    def has_spam_domain(obj):
```

Comment:
I think this should also be a method on the `SpamMixin` class because it is describing whether a resource have contents that contains spam domains.

We could modify this static method to a `content_contains_spam_domain` method that takes in only `content` and returns a boolean. And then we can have the `has_spam_domain` method on the `SpamMixin` to call this static method.